### ISSUE TITLE
Offline Visualization

### ISSUE THREAD

**Issue by AFOliveira** [riscv-software-src/riscv-unified-db#79]
As of now all images are hosted in kroki.io, is there any specific reason for that. I do realize this is low priority, but for future purposes I believe that we should try to avoid to rely on this external tools, right? 
Offline visualization:
![image](https://github.com/user-attachments/assets/313c1c34-48ef-4c34-8c08-b0ddcddca0c1)

**Comment by dhower-qc** [comment]
Yes, we should. The only reason we're using that is because the only public plugin for wavedrom in Asciidoctor.js (needed by Antora) uses Kroki. To fix this, we'll need to write a custom plugin.

**Comment by dhower-qc** [comment]
We are using wavedrom. The issue is that there are multiple asciidoctor plugins for wavedrom that can only be used with certain tools.

riscv-isa-manual uses the Ruby backend ('asciidoctor'), and the wavedrom plugin for that generates and stores wavedrom images locally. riscv-unified-db also uses the Ruby backend to create PDFs. 

However, the tool being used to create the html ISA manual out riscv-unified-db is Antora, which can only use a JavaScript backend ('asciidoctor-js') of Asciidoctor. The wavedrom plugin for asciidoctor-js is what generates those on-demand image links using Kroki. (For what it's worth, Bill Traynor from RVI has an Antora version of riscv-isa-manual, and it has the same on-demand image issue).

**Comment by AFOliveira** [comment]
Got it! Thanks for the explanation!

**Comment by dhower-qc** [comment]
@drom, are you aware of any alternatives that will help get static wavedrom images in pages generated from Antora? Right now, we are using the Kroki plugin that generates the image at page load time instead of page creation time.

**Comment by drom** [comment]
I never used Antoria. Does it allow running some JS inside the page? we can render diagrams dynamically.
